### PR TITLE
[improve](array-funcs)support arrays_overlap with invertedIndex

### PR DESCRIPTION
**Pull request by amorynan**
support arrays_overlap with invertedIndex 

## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run p0

**Review by zzzxl1993 — APPROVED** [review]
LGTM

File: be/src/vec/functions/array/function_arrays_overlap.h [hunk 1]
```diff
@@ -128,6 +129,83 @@ class FunctionArraysOverlap : public IFunction {
...
+            std::shared_ptr<roaring::Roaring> single_res = std::make_shared<roaring::Roaring>();
+            RETURN_IF_ERROR(InvertedIndexQueryParamFactory::create_query_value(
+                    nested_param_type, &nested_query_val, query_param));
+            Status st = iter->read_from_inverted_index(
+                    data_type_with_name.first, query_param->get_value(),
+                    segment_v2::InvertedIndexQueryType::EQUAL_QUERY, num_rows, single_res);
+            if (st.code() == ErrorCode::INVERTED_INDEX_NO_TERMS) {
+                // if analyzed param with no term, we do not filter any rows
+                // return all rows with OK status
+                roaring->addRange(0, num_rows);
```
Comment (airborne12) [inline be/src/vec/functions/array/function_arrays_overlap.h:195]: can fast execute here, if return code is INVERTED_INDEX_NO_TERMS, we can return now.

File: regression-test/suites/inverted_index_p0/test_array_contains_with_inverted_index.groovy [hunk 2]
```diff
@@ -76,4 +76,18 @@ suite("test_array_contains_with_inverted_index"){
...
+    order_qt_sql """ select * from tai where arrays_overlap(inventors, array('s', 't')) order by id; """
+    order_qt_sql """ select * from tai where arrays_overlap(inventors, array('s', 't')) and apply_date = '2017-01-01' order by id; """
+    order_qt_sql """ select * from tai where arrays_overlap(inventors, array('s', 't')) and apply_date = '2019-01-01' order by id; """
+    order_qt_sql """ select * from tai where arrays_overlap(inventors, array('s', 't')) or apply_date = '2017-01-01' order by id; """
+    order_qt_sql """ select * from tai where !arrays_overlap(inventors, array('s', 't')) order by id; """
+    order_qt_sql """ select * from tai where !arrays_overlap(inventors, array('s', 't')) and apply_date = '2017-01-01' order by id; """
+    order_qt_sql """ select * from tai where !arrays_overlap(inventors, array('s', 't')) and apply_date = '2019-01-01' order by id; """
+    order_qt_sql """ select * from tai where !arrays_overlap(inventors, array('s', 't')) or apply_date = '2017-01-01' order by id; """
+    order_qt_sql """ select * from tai where (arrays_overlap(inventors, array('s', 't')) and apply_date = '2017-01-01') or apply_date = '2019-01-01' order by id; """
+
```
Comment (airborne12) [inline regression-test/suites/inverted_index_p0/test_array_contains_with_inverted_index.groovy:91]: may be more test here, like null value etc.

Comment (amorynan) [reply regression-test/suites/inverted_index_p0/test_array_contains_with_inverted_index.groovy]: done!

**Comment by amorynan** [comment]
run buildall

**Review by airborne12 — APPROVED** [review]
LGTM

**Pull request merged**